packages/devextreme/js/ui/drop_down_button.js

```diff
@@ -117,14 +115,23 @@ const DropDownButton = Widget.inherit({
         this._createItemClickAction();
         this._createActionClickAction();
         this._createSelectionChangedAction();
-        this._initDataSource();
+        this._initDataController();
         this._compileKeyGetter();
         this._compileDisplayGetter();
-        this._itemsToDataSource(this.option('items'));
         this._options.cache('buttonGroupOptions', this.option('buttonGroupOptions'));
         this._options.cache('dropDownOptions', this.option('dropDownOptions'));
     },
 
+    _initDataController() {
+        const dataSource = this.option('dataSource');
+        let key;
+
+        if(!dataSource) {
+            key = this._getKey();
+        }
+        this._dataController = new DataController(dataSource ?? this.option('items'), key);
```

Comment:
I suggest u making the second parameter object `{ key: this._getKey() }` and always pass it.
It'll make this code more clear.
Ideologically i think we should always pass all possible-to-use props to DataController and already DataContoller should decide when to use it and when not.